src/cp/cp.rs

```diff
@@ -780,7 +784,21 @@ fn copy_attribute(source: &Path, dest: &Path, attribute: &Attribute) -> CopyResu
             let metadata = fs::metadata(source).context(context)?;
             fs::set_permissions(dest, metadata.permissions()).context(context)?;
         },
-        Attribute::Timestamps => return Err(Error::NotImplemented("preserving timestamp not implemented".to_string())),
+        Attribute::Timestamps => {
+            let meta = fs::metadata(source)?;
+            let modified = meta.modified()?;
+            let accessed = meta.accessed()?;
+            let src_path = CString::new(source.as_os_str().to_str().unwrap()).unwrap();
+            let dest_path = CString::new(dest.as_os_str().to_str().unwrap()).unwrap();
+            unsafe {
+                let mut stat: libc::stat = mem::zeroed();
+                libc::stat(src_path.as_ptr(), &mut stat);
```

Comment:
You should be able to use the `filetime` crate to get and set the times; that should be more portable, and work on Windows and Redox.